CI/package-deps-windows.ps1

```diff
@@ -0,0 +1,159 @@
+Param(
+    [Switch]$Help = $(if (Test-Path variable:Help) { $Help }),
+    [Switch]$Quiet = $(if (Test-Path variable:Quiet) { $Quiet }),
+    [Switch]$Verbose = $(if (Test-Path variable:Verbose) { $Verbose }),
+    [String]$BuildDirectory = "build"
+)
+
+##############################################################################
+# Windows OBS deps package script
+##############################################################################
+#
+# This script contains all steps necessary to:
+#
+#   * Build OBS required dependencies
+#   * Create 64-bit and 32-bit variants
+#
+# Parameters:
+#   -Help                   : Print usage help
+#   -Quiet                  : Suppress most build process output
+#   -Verbose                : Enable more verbose build process output
+#   -BuildDirectory         : Directory to use for builds
+#                             Default: Win64 on 64-bit systems
+#                                      Win32 on 32-bit systems
+#
+##############################################################################
+
+$ErrorActionPreference = "Stop"
+
+function Package-OBS-Deps-Main {
+    $CheckoutDir = git rev-parse --show-toplevel
+    $ProductName = "windows-deps"
+    $DepsBuildDir = "${CheckoutDir}\windows\obs-win-deps"
+
+    . "${CheckoutDir}\CI\include\build_support_windows.ps1"
+
+    Write-Status "Create Windows OBS deps package"
+
+    Ensure-Directory "${CheckoutDir}"
+
+    if (!$CurrentDate) {
+        $CurrentDate = Get-Date -UFormat "%Y-%m-%d"
+    }
+    $FileName = "${ProductName}-${CurrentDate}.tar.gz"
+    $CrossDirBase = "${CheckoutDir}\windows\obs-cross-deps"
+    $NativeDirBase = ""
+
+    if (Test-Path "${CheckoutDir}\windows\obs-native-deps" -PathType "Container") {
+        $NativeDirBase = "${CheckoutDir}\windows\obs-native-deps"
+    } elseif (Test-Path "${CheckoutDir}\windows_native_build_temp" -PathType "Container") {
+        $NativeDirBase = "${CheckoutDir}\windows_native_build_temp"
+    }
+
+    if (!(Test-Path "${CrossDirBase}\x86" -PathType "Container")) {
+        Caught-Error "Missing cross-compiled build in ${CrossDirBase}\x86"
+    } elseif (!(Test-Path "${CrossDirBase}\x86_64" -PathType "Container")) {
+        Caught-Error "Missing cross-compiled build in ${CrossDirBase}\x86_64"
+    } elseif (!(Test-Path "${NativeDirBase}\win32" -PathType "Container")) {
+        Caught-Error "Missing native build in ${NativeDirBase}\win32"
+    } elseif (!(Test-Path "${NativeDirBase}\win64" -PathType "Container")) {
+        Caught-Error "Missing native build in ${NativeDirBase}\win64"
+    }
+
+    Remove-ItemIfExists "${DepsBuildDir}"
+    Ensure-Directory "${DepsBuildDir}\win32"
+    Ensure-Directory "${DepsBuildDir}\win64"
+    cd ..
+
+    $Packages = @(
+        [pscustomobject]@{Arch='x86'; WinArch='win32'}
+        [pscustomobject]@{Arch='x86_64'; WinArch='win64'}
+    )
+    Foreach ($Package in $Packages) {
+        if ($Package.Arch -eq "x86") {
```

Comment:
I've come to prefer this pattern:

```
    $QtDirectory = "${CheckoutDir}/../obs-build-dependencies/Qt_${WindowsQtVersion}/msvc2019$(if (${BuildArch} -eq "64-bit") { "_64" })"
    $DepsDirectory = "${CheckoutDir}/../obs-build-dependencies/dependencies${WindowsDepsVersion}/win$(if (${BuildArch} -eq "64-bit") { "64" } else { "32" })"
    $CefDirectory = "${CheckoutDir}/../obs-build-dependencies/cef_binary_${WindowsCefVersion}_windows$(if (${BuildArch} -eq "64-bit") { "64" } else { "32" })_minimal"
```
Matter of taste probably.

Reply:
I find that inline `if` or ternary statements are a bit jarring or unclear.  Admittedly, in such cases, I prefer using an intermediate variable.